packages/extras/src/lib/components/Gizmo/Gizmo.svelte

```diff
@@ -0,0 +1,343 @@
+      clickTarget.style.top = ''
+    } else {
+      clickTarget.style.bottom = ''
+      clickTarget.style.top = '0'
+    }
+
+    clickTarget.style.height = `${size}px`
+    clickTarget.style.width = `${size}px`
+  }
+
+  let posX: Sprite
+  let posY: Sprite
+  let posZ: Sprite
+  let negX: Sprite
+  let negY: Sprite
+  let negZ: Sprite
+
+  const targetPosition = new Vector3()
+  const targetQuaternion = new Quaternion()
+  const currentQuaternion = new Quaternion()
+  const finalQuaternion = new Quaternion()
+  let radius = 0
+
+  let animating = false
+  const mouse = new Vector2()
+  const raycaster = new Raycaster()
+
+  const handleClick = (event: MouseEvent) => {
+    if (animating) {
+      return
+    }
+
+    // Raycasting is done manually.
+    const rect = clickTarget.getBoundingClientRect()
+    const offsetX = rect.left + (clickTarget.offsetWidth - size)
+    const offsetY = rect.top + (clickTarget.offsetHeight - size)
+    mouse.x = ((event.clientX - offsetX) / (rect.right - offsetX)) * 2 - 1
+    mouse.y = -((event.clientY - offsetY) / (rect.bottom - offsetY)) * 2 + 1
+
+    raycaster.setFromCamera(mouse, orthoCam)
+
+    const intersects = raycaster.intersectObjects([posX, posY, posZ, negX, negY, negZ])
+
+    if (intersects.length > 0) {
+      const intersection = intersects[0]
+      const object = intersection.object
+      const targetPos = object.userData.targetPosition as [number, number, number]
+      const targetEuler = object.userData.targetEuler as [number, number, number]
+
+      radius = camera.current.position.distanceTo(centerVec)
+      targetPosition
+        .set(...targetPos)
+        .multiplyScalar(radius)
+        .add(centerVec)
+      targetQuaternion.setFromEuler(new Euler(...targetEuler))
+
+      const dummy = new Object3D()
+      dummy.position.copy(centerVec)
+
+      dummy.lookAt(camera.current.position)
+      currentQuaternion.copy(dummy.quaternion)
+
+      dummy.lookAt(targetPosition)
+      finalQuaternion.copy(dummy.quaternion)
+
+      animating = true
+
+      return true
+    } else {
+      return false
+    }
+  }
+
+  onMount(() => {
+    renderer.domElement.parentElement?.appendChild(clickTarget)
+    clickTarget.addEventListener('click', handleClick)
+  })
+
+  onDestroy(() => {
+    renderer.domElement.parentElement?.removeChild(clickTarget)
+    clickTarget.removeEventListener('click', handleClick)
+  })
+
+  // Used to test which axis (pos or neg) are closer to the camera.
+  const point = new Vector3()
+  let p = [0, 0, 0]
+  // The rotation of the main camera, used to rotate the gizmo.
+  const quaternion = new Quaternion()
+  let q: [number, number, number, number] = [0, 0, 0, 0]
+
+  useFrame(({ camera }, delta) => {
+    point.set(0, 0, 1).applyQuaternion(camera.current.quaternion)
+    p = [point.x, point.y, point.z]
+    quaternion.copy(camera.current.quaternion).invert()
+    q = [quaternion.x, quaternion.y, quaternion.z, quaternion.w]
```

Comment:
in this case I'd recommend:
 ```suggestion
   root.quaternion.copy(camera.current.quaternion).invert()
```
and removing `quaternion={q}` below as this can be more expensive when in the frameloop